lib/spot-checker.js

```diff
@@ -86,9 +96,12 @@ export default class SpotChecker {
    * @param {string} args.address
    * @param {string} args.cid
    * @param {object} args.stats
-   * @param {number} [args.maxByteLength]
+   * @param {'all' | 'block'} args.dagScope
+   * @param {number} [args.maxByteLength=0] - Maximum bytes to download before stopping retrieval (applies to full retrievals only)
    * */
-  async fetchCAR ({ protocol, address, cid, stats, maxByteLength }) {
+  async fetchCAR ({ protocol, address, cid, stats, dagScope, maxByteLength = 0 }) {
+    const retrievalStats = dagScope === 'all' ? stats.fullRetrieval : stats.blockRetrieval
```

Comment:
Let's move this condition outside of `fetchCAR` to the caller.

```js
    await this.fetchCAR({
      protocol: provider.protocol,
      address: provider.address,
      cid: task.cid,
      dagScope: 'block',
      stats.blockRetrieval,
      maxByteLength
    })
   
    await this.fetchCAR({
      protocol: provider.protocol,
      address: provider.address,
      cid: task.cid,
      dagScope: 'all',
      fullRetrieval,
      maxByteLength
    })
```

Reply:
When we move that logic outside of `fetchCAR`, it's easier to undertand which `stats` properties will be populated by a `fetchCAR` code, and it will be easier to add a check using a different combinations of fetchCAR parameters without having to modify fetchCAR.

Reply:
Thank you for your feedback 🙏🏻 I've addressed changes in https://github.com/CheckerNetwork/spark-spot-check/pull/6/commits/b5068e3e28f1e841701b98921cf48ef9d2a46b5a